dev/com.ibm.ws.classloading.bells/src/com/ibm/ws/classloading/bells/internal/Bell.java

```diff
@@ -516,10 +543,27 @@ private static Class<?> findClass(final String implClass, final Library library,
         return service;
     }
 
-    private static Object createService(final Class<?> serviceType, final String libID, final URL fileUrl) {
+    private static Object createService(final Class<?> serviceType, final String libID, final URL fileUrl, final Map<String, String> properties) {
         Object service = null;
         try {
-            service = serviceType.newInstance();
+            if (properties == null) {
+                service = serviceType.newInstance();
+            } else {
+                // Inject a shallow clone of the properties map
+                final Constructor<?> singleArgCtor;
+                final Method updateMethod;
+                if ((singleArgCtor = getConstructor(serviceType, java.util.Map.class)) != null) {
+                    service = singleArgCtor.newInstance(((HashMap<String, String>) properties).clone());
```

Comment:
Yes, I'll document the map is unmodifiable, because I'm having trouble enforcing this contract in the code using a unmodifiableMap. Having trouble passing a unmodifiableMap(hashMap<s,s>) over a parameter of type Map<s,s>:

> The following error occurred: java.lang.ClassCastException: java.util.Collections$UnmodifiableMap incompatible with java.util.HashMap
> 	at com.ibm.ws.classloading.bells.internal.Bell.createService(Bell.java:559)
> 	at [internal classes]